src/acl.c

```diff
@@ -1458,16 +1459,30 @@ void addAuthErrReply(client *c, robj *err) {
  *
  * The return value is AUTH_OK on success (valid username / password pair) & AUTH_ERR otherwise. */
 static int checkPasswordBasedAuth(client *c, robj *username, robj *password) {
+    int result;
+
     if (ACLCheckUserCredentials(username, password) == C_OK) {
         user *user = ACLGetUserByName(username->ptr, sdslen(username->ptr));
         clientSetUser(c, user, 1);
         moduleNotifyUserChanged(c);
-        return AUTH_OK;
+        result = AUTH_OK;
     } else {
         addACLLogEntry(c, ACL_DENIED_AUTH, (c->flag.multi) ? ACL_LOG_CTX_MULTI : ACL_LOG_CTX_TOPLEVEL, 0, username->ptr,
                        NULL);
-        return AUTH_ERR;
+        result = AUTH_ERR;
+    }
+
+    {
+        ValkeyModuleAuthenticationInfo info = VALKEYMODULE_AUTHENTICATIONINFO_INITIALIZER_V1;
+        info.client_id = c->id;
+        info.username = username->ptr;
+        info.module_name = NULL;
+        info.result = result == AUTH_OK ? VALKEYMODULE_AUTH_RESULT_GRANTED
+                                        : VALKEYMODULE_AUTH_RESULT_DENIED;
+        moduleFireServerEvent(VALKEYMODULE_EVENT_AUTHENTICATION_ATTEMPT, 0, &info);
```

Comment:
This code is not large, but it's duplicated. If we change V1 to V2 in the future, we'll need to remember to change in both places.

Let's move this and the duplicated code in `checkModuleAuthentication` to a new function in `module.{c,h}`? e.g. `moduleFireAuthenticationEvent(client_id, username, module_name, is_granted)`.

Then, acl.c doesn't need to include valkeymodule.h. It's enough to include module.h.